hw/ip/otbn/code-snippets/otbn_bn_mul64.S

```diff
@@ -0,0 +1,548 @@
+/* Copyright lowRISC contributors. */
```

Comment:
The advice is to leave it as BSD licensed, and add a "Copyright lowRISC contributors" line above the Chromium copyright line.

Reply:
Oh no! I think that will confuse Sam's tool yet further! I've pushed up a version that I think is what you're describing, but with a little bit of explanatory text for some poor soul who comes on the file in the future: does this look right to you?

Reply:
@lenary: Any idea how best to convince the licence checker to leave this alone?

Reply:
You explicitly add it to the `util/licence-checker.hjson` as an exception, with a comment as to why.

Reply:
Ahah! Thanks.

Reply:
I'm not a fan of the license header split into two comments, and would also remove the explanatory comment to go with what exactly Alex proposed. (We did something similar in Ibex at https://github.com/lowRISC/ibex/blob/master/rtl/ibex_core.sv.)

```
Copyright lowRISC Contributors.
Copyright 2010 The Chromium OS Authors. 
All rights reserved.
```


Alternatively, keep the original short license header (https://chromium.googlesource.com/chromiumos/platform/ec/+/refs/heads/cr50_stab/chip/g/dcrypto/dcrypto_bn.c) and add a separate LICENSE.dcrypto file to this directory with the original license text. This should help us if we split this file into multiple smaller ones.

@asb up to you ultimately.

Reply:
I think the original short-form header (modified to mention LICENSE.dcrypto, and with the lowRISC copyright line prepended) would be fine, and might better prepare us for the fact there could be other snippets with the same license. I think separate to that, there is some value in explaining where the code came from (e.g. referencing dcrypto_bn.c and giving a link). This can be a short comment below the license header.

I'd also be fine with a version that's roughly what Rupert currently has, but tweaked so the "Copyright lowRISC contributors." bit goes directly before the Chromium OS Authors copyright line (and perhaps the description tweaked so it's obvious what's going on there).

Reply:
Cool, I'll split it out into a LICENCE.dcrypto and add a pointer.

Reply:
I've just pushed something. Note that it now claims to be releasing the code under the Apache licence. I believe that you can do BSD -> Apache, but my original proposals were trying to avoid that (so I didn't have to think about it). @asb / @imphil: Is this the sort of thing you meant?

Reply:
I meant taking the short-form header from dcrypto_bn.c and modifying like below:

```
/* Copyright lowRISC Contributors.
 * Copyright 2016 The Chromium OS Authors. All rights reserved.
 * Use of this source code is governed by a BSD-style license that can be
 * found in the LICENSE.dcrypto file.
 */
```


Then adding a comment below that header that's something like:
"An example of how to use the 64-bit multiplier to do 256-bit multiplications. Derived from code in https://chromium.googlesource.com/chromiumos/platform/ec/+/refs/heads/cr50_stab/chip/g/dcrypto/dcrypto_bn.c"

The latter isn't a licensing requirement, but it useful background.

Reply:
Ah, right. That makes much more sense than what I thought you were saying! Done.